test/distributed/algorithms/quantization/test_quantization.py

```diff
@@ -117,23 +128,46 @@ def test_all_to_all_fp16(self):
                 dtype=torch.float32,
                 qtype=DQuantType.FP16)
 
+        @requires_nccl()
+        @sandcastle_skip_if(BACKEND != "nccl", "Only nccl backend supports all_to_all_fp16")
+        @skip_if_lt_x_gpu(int(os.environ["WORLD_SIZE"]))
+        @skip_if_rocm
+        def test_all_to_all_bfp16(self):
+            store = dist.FileStore(self.file_name, self.world_size)
+            dist.init_process_group(store=store, rank=self.rank, world_size=self.world_size, backend='nccl')
+            device = torch.device(f"cuda:{self.rank}")
+            group = list(range(0, self.world_size))
+            group_id = dist.new_group(range(self.world_size))
+            rank_to_GPU = self._init_multigpu_helper()
+            self._test_all_to_all(
+                group,
+                group_id,
+                self.rank,
+                cuda=True,
+                rank_to_GPU=rank_to_GPU,
+                dtype=torch.float32,
+                qtype=DQuantType.BFP16)
+
         def _test_all_gather(
                 self, group, group_id, rank, cuda=False, rank_to_GPU=None, dtype=torch.float, qtype=None):
             for dest in group:
-                tensor = _build_tensor(dest + 1, rank, dtype=dtype)
-                tensors = [_build_tensor(dest + 1, -1, dtype=dtype) for i in group]
-                expected_tensors = [_build_tensor(dest + 1, i, dtype=dtype) for i in group]
-                if (qtype is not None):
-                    allgather = quant.auto_quantize(dist.all_gather, qtype, quant_loss=None)
-                else:
-                    allgather = dist.all_gather
+                tensor = _build_tensor([dest + 1, dest + 1, dest + 1], rank, dtype=dtype)
```

Comment:
is it necessary to distinguish between fp16 and bfp16 when building tensor inputs? why don't we just all use 2 dim tensor?

Reply:
This is the datatype (dtype) I just wanted to make sure that the input tensor is FP32.

Reply:
what does it mean? I was referring to the IF branch below that you are creating a two dimensional tensor for BFP16 case, but for other cases we are using three dimensional tensor, why not we just creating two dimensional tensor for all cases?